jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/SingleThreadedLogger.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.plugins.common;
+
+import com.google.common.util.concurrent.Futures;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.function.Consumer;
+
+/**
+ * Keeps all log messages in a sequential, deterministic order along with an additional footer that
+ * always appears below log messages.
+ */
+class SingleThreadedLogger {
```

Comment:
This class name doesn't really describe the class well.  This logger is only appropriate for terminals that support ANSI escape codes.

Reply:
Hmm what would you recommend as the name?

Reply:
`SureFootedLogger` :-)

More seriously, the key here is that the logger uses ANSI escapes to maintain a footer element. `AnsiFooterLogger`?

Reply:
Hmm, this is more of a logger than logs messages along with a footer that is always at the bottom and `AnsiFooterLogger` seems to point to it being just logging a footer - how about `AnsiLoggerWithFooter`?

Reply:
I think `AnsiLoggerWithFooter` works.

I'd just like to say in the Javadoc that the class is intended for the case where both of the two log destination channels are the same console. (For example, it isn't really for `messageLogger` to log to a file or a database, while the footer is going out to stdout.)